src/main/webapp/front-end/src/components/Register.jsx

```diff
@@ -2,22 +2,45 @@ import React from 'react';
 import {
   Grid, 
   PageHeader,
-  Button  
+  Button
 } from 'react-bootstrap';
-import { Link } from 'react-router-dom';
-import EmailFormGroup from './account_forms/EmailFormGroup.jsx'
+import { connect } from 'react-redux';
+import { Link, Redirect } from 'react-router-dom';
+import { signupEmailChanged, signupPasswordChanged, signupClearInputs, createUser, createUserSuccess, createUserError } from '../actions.js'
+import { tryLogin } from '../utilities.js';
+import EmailFormGroupContainer from './account_forms/EmailFormGroup.jsx'
 import PasswordFormGroup from './account_forms/PasswordFormGroup.jsx'
 
-export default class Register extends React.Component {
+class Register extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redirectToReferrer: false
+    }
+
+    this.onSubmit = (e) => {
+      e.preventDefault();
+      this.props.onSubmit(this.props.email, this.props.password)
+        .then(response => {
+          this.setState({ redirectToReferrer: response })
```

Comment:
Sure, I was just sticking close to react router example code here -- I think in certain cases like this one local state makes sense, if it doesn't touch affect the rest of the codebase